lib/autocomplete.js

```diff
@@ -0,0 +1,90 @@
+const _ = require("lodash");
+const aws = require("aws-sdk");
+const parsers = require("./parsers");
+const core = require("./core");
+const consts = require("./consts.json");
+
+/**
+ * Maps parameters passed to Autocomplete function to an object with parsed values.
+ *
+ * @param params: [ { value: any, name: string, type: string } ]
+ * Array of values passed to autocomplete function as action parameters.
+ *
+ * @returns {*} - object with params names as keys and parsed values
+ */
+function parseActionParameters(params) {
+  return params.reduce((acc, {
+    value, name, type, valueType,
+  }) => ({
+    ...acc,
+    [name]: parsers.resolveParser(type || valueType)(value),
+  }), {});
+}
+
+function handleInput(awsService, actionParams, pluginSettings) {
+  const [params, settings] = [actionParams, pluginSettings].map(parseActionParameters);
+  const client = core.getServiceInstance(awsService, params, settings);
+  return { params, settings, client };
+}
+
+function simpleItemsFilter(autocompleteItems, query) {
+  let result = autocompleteItems;
+  if (query) {
+    const qWords = query.split(/[. ]/g).map((word) => word.toLowerCase());
```

Comment:
`queryWords` - we don't save much time, not memory by limiting outselves to `q`.